src/machine/lib_machine/mod.rs

```diff
@@ -0,0 +1,632 @@
+use std::cmp::Ordering;
```

Comment:
Why the change from `lib_machine.rs` to `lib_machine/mod.rs`?

My personal preference would be the former. 
Depending on the editor I am using I either have a lot of indistinguishable `mod.rs` editor tabs or `<mod_name>\mod.rs` rather than `<mod_name>.rs` editor tabs (i.e. a longer title) which I find annoying to navigate.

Reply:
Maybe we should enable either the [`clippy::mod_module_files`](https://rust-lang.github.io/rust-clippy/master/index.html#mod_module_files) or [`clippy::self_named_module_files`](https://rust-lang.github.io/rust-clippy/master/index.html#self_named_module_files) lint to enforce a consistent style for the project?

Reply:
I did this mostly to have the tests for the module in a different file, which makes compilation of tests much faster and helped me iterate. This doesn't necessarily need a module directory, but it makes it a bit more organized. I can change it back. Personally I like having the tests separated like this, but this would be a really big change for the whole project so I don't think it is appropriate to do it in this PR, if we decide to actually migrate.

Reply:
I just forgot to rename it, it should indeed be `lib_machine/tests.rs`.